### PR TITLE
Add ZendMM internals, lifecycle and how to hook into docs

### PR DESCRIPTION
**Pull request by realFlowControl**
Hey there 👋 

this PR will add a missing documentation about the ZendMM internal design, as well as a section about the lifecycle of the Zend Memory Manager and an example of how to hook into it.

### CODE REVIEW

File: Book/php7/memory_management/zend_memory_manager.rst [hunk 1]
```diff
@@ -220,10 +220,137 @@ Beware however:
...
++++++
+
+A page is 4096 bytes in size and can either hold a bin (for small allocations) or be part of a large allocation. What is
+in it can be found in the map of the chunk the page belongs to.
+
+Bins
+++++
+
+Small allocations are grouped together in bins. Bin sizes are predefined and come in 30 different sizes (8, 16, 24, 32,
+... 3072). A bin holds same sized values and is linked from the heap directly.
```
Comment (Girgias) [inline Book/php7/memory_management/zend_memory_manager.rst:283]: "8, 16, 24" what? I suppose bytes?

File: Book/php7/memory_management/zend_memory_manager.rst [hunk 2]
```diff
@@ -220,10 +220,137 @@ Beware however:
...
+
+A bin can consist of multiple pages. Example: There is a bin that holds elements the sizes 257 bytes to 320 bytes which
+occupies 5 pages and therefore has room for 64 elements of that size.
```
Comment (Girgias) [inline Book/php7/memory_management/zend_memory_manager.rst:286]:
```suggestion
A bin can consist of multiple pages. Example: There exists a bin that holds elements ranging from 257 bytes to 320 bytes which occupies 5 pages, and therefore has room for 64 (derived from 4096*5/320) elements of that size.
```

File: Book/php7/memory_management/zend_memory_manager.rst [hunk 3]
```diff
@@ -220,10 +220,137 @@ Beware however:
...
+
+Allocation categories
+---------------------
+
+Small allocations
++++++++++++++++++
+
+Allocations smaller than or up to 3072 bytes are organised in bins.
```
Comment (Girgias) [inline Book/php7/memory_management/zend_memory_manager.rst:294]:
```suggestion
Allocations less or equal than 3072 bytes are organised in bins.
```

**Review by Girgias — COMMENTED** [review]
Thank you! I don't know much about ZendMM but some wording comments. :)

File: Book/php7/memory_management/zend_memory_manager.rst (hunk 1 above)
Comment (realFlowControl) [reply Book/php7/memory_management/zend_memory_manager.rst]: Correct, I fixed it

File: Book/php7/memory_management/zend_memory_manager.rst [hunk 4]
```diff
@@ -220,10 +220,137 @@ Beware however:
 What you must remember is that ZendMM leak tracking is a nice bonus tool to have, but it does not replace a
 :doc:`true C memory debugger <./memory_debugging>`.
 
+Lifecycle
+*********
+
+PHP will call the ``start_memory_manager()`` function during it's startup phase, specifically when the PHP process is
```
Comment (kocsismate) [inline Book/php7/memory_management/zend_memory_manager.rst:226]:
```suggestion
PHP will call the ``start_memory_manager()`` function during its startup phase, specifically when the PHP process is
```

File: Book/php7/memory_management/zend_memory_manager.rst [hunk 5]
```diff
@@ -220,10 +220,137 @@ Beware however:
...
 **********************
 
-.. todo:: todo
+The root of the ZendMM is the ``_zend_mm_heap`` struct (as defined in `Zend/zend_alloc.c
+<https://github.com/php/php-src/blob/c3b910370c5c92007c3e3579024490345cb7f9a7/Zend/zend_alloc.c#L239>`__) which will be
+created for every request during request init and stored in the ``alloc_globals->mm_heap``. This heap also comes with
+the first chunk that is allocated with it. Chunks are then subdivided into pages. Smaller allocations are stored in bins
+which may fit into one page but some also span multiple pages.
+
+Interal memory organisation
```
Comment (realFlowControl) [inline Book/php7/memory_management/zend_memory_manager.rst:251]:
```suggestion
Internal memory organisation
```

**Comment by arnaud-lb** [comment]
Looks good to me, thank you!

**Pull request merged**